### ISSUE TITLE
newbie question

### ISSUE THREAD

**Issue by YaPaY** [rbreaves/kinto#550]
Thanks for such a wonderful tool!

it is really easy question. I am using pop OS 21.04 with tile windows but after install the script I lost super key :)

which key assigned as super now?

**Comment by rbreaves** [comment]
Cmd-Space (1st key left of the spacebar)

**Comment by YaPaY** [comment]
thank you very much for quick help but Cmd-Space or any other Space+ system keys combinations does not work. Should I uninstall and install again?

**Comment by rbreaves** [comment]
I'd suggest looking at the keyboard shortcut GUI first, may just want to reset the launcher there. I doubt a reinstall would fix it if it didn't work the 1st time. Maybe I never made sure that combo worked, although I do like Pop OS so kinda surprising to me that it wouldn't work out of the box as I know I had tested things on it not that long ago.

I do need a CI workflow to run automated tests though.

**Comment by YaPaY** [comment]
after uninstall and install did not change anything but after added your code now super key is become ctrl (I am using apple Magic Keyboard)

**Comment by rbreaves** [comment]
True, the Super key moves to the Ctrl position as that is what happens w/ Kinto.

You'd need to set it to Control if you really want it in the same place - but this is why Kinto does what it does.. it tries to remap Super to Cmd-Space which is why I am telling you to reset it via the GUI so you can make sure you get both keys mapped to it. The overlay is how it is mapped by default, but the Keyboard shortcut GUI will map it a little differently so that it is a combo key instead.

**Comment by rbreaves** [comment]
To unset the overlay key just run this. No need to do a reinstall.

```
gsettings set org.gnome.mutter overlay-key ''

```

**Comment by rxng** [comment]
This is terrible, I am on popos, and after installing, super key stopped working. I then uninstalled Kinto and now the super is gone and will not comeback. Can you help us get the super key back please?

**Comment by rbreaves** [comment]
@rxng that is odd, the uninstall ought to bring it back but it’s literally the only OS level remap that happens these days as it would otherwise trip up many shortcuts.

And it didn’t stop working - it just gets disabled in favor of the macOS combo Cmd-Space as is explained in many places. This thread also already has the answer for resetting it.

```
gsettings set org.gnome.mutter overlay-key 'super'
```

**Comment by rbreaves** [comment]
Looking at my uninstaller code I pretty clearly restore the Super key back.. so unless you’re on a DE that isn’t one of these.. but even then.. I do different kinds of resets or restores for other DEs still - this is just what ought to apply to gnome on pop_os.

```
if [[ $dename == "gnome" || $dename == "budgie" || $dename == "mate" ]]; then
		echo -e "\nWill still be restoring the overlay key"
		echo -e "gsettings set org.gnome.mutter overlay-key 'super'\n"
		gsettings set org.gnome.mutter overlay-key 'super'
fi
```

**Comment by canadaduane** [comment]
When I run `sudo keyd -m` (i.e. monitor mode for keyboard evdev devices) and press Cmd-Space these are the events I see:

```
py-evdev-uinput: rightcontrol down
py-evdev-uinput: rightcontrol up
py-evdev-uinput: alt down
py-evdev-uinput: f1 down
py-evdev-uinput: f1 up
py-evdev-uinput: rightcontrol down
py-evdev-uinput: rightcontrol up
py-evdev-uinput: alt up
```

Count me confused!

**Comment by rbreaves** [comment]
@canadaduane According to this gnome and kde use Alt-F1, not sure if my installer sets that up as an alternative or if that is a built in alternative. I'd have to look and it is late.

What your monitor is picking up on is the physical keyboard PLUS the virtual udex/input device. Only your key presses from the virtual is reaching the actual OS. All physical keys are consumed by xkeysnail and either repeated as is or transformed accordingly. Although I am not sure why the spacebar key does not appear in your log there - that is the only part that confuses me. Perhaps the last key in that combo gets cancelled out so quickly that the monitor literally does not have time to pickup on it.

https://github.com/rbreaves/kinto/blob/master/linux/kinto.py#L458
```
    K("RC-Space"): K("Alt-F1"),                   # Default SL - Launch Application Menu (gnome/kde)
```

**Comment by RobSjaras** [comment]
> @canadaduane According to this gnome and kde use Alt-F1, not sure if my installer sets that up as an alternative or if that is a built in alternative. I'd have to look and it is late.
> 
> What your monitor is picking up on is the physical keyboard PLUS the virtual udex/input device. Only your key presses from the virtual is reaching the actual OS. All physical keys are consumed by xkeysnail and either repeated as is or transformed accordingly. Although I am not sure why the spacebar key does not appear in your log there - that is the only part that confuses me. Perhaps the last key in that combo gets cancelled out so quickly that the monitor literally does not have time to pickup on it.
> 
> https://github.com/rbreaves/kinto/blob/master/linux/kinto.py#L458
> 
> ```
>     K("RC-Space"): K("Alt-F1"),                   # Default SL - Launch Application Menu (gnome/kde)
> ```

Hi, sorry to bump, but I just wanted to say thanks for both Kinto.sh and also the thread here!

And for future readers.. what helped me fix my issues in Pop_OS! was to set the keyboard shortcut for `Launch and switch applications` to _Command+Space_ which should then input the above-mentioned `Alt-F1` as the input keys for the shortcut, and bam! Now it should work :)

**Comment by NurdinDev** [comment]
I found this way to change the default key of launcher Pop_os 22.04

`dconf write /org/gnome/shell/extensions/pop-shell/activate-launcher "['<Alt>F1']"`

>Alt-F1 = Cmd-space (I don't know why)
